core/services/vrf/listener_v2.go

```diff
@@ -662,6 +713,79 @@ func (lsn *listenerV2) processRequestsPerSubBatch(
 	return processed
 }
 
+// enqueueForceFulfillment enqueues a forced fulfillment through the
+// VRFOwner contract. It estimates gas again on the transaction due
+// to the extra steps taken within VRFOwner.fulfillRandomWords.
+func (lsn *listenerV2) enqueueForceFulfillment(
+	ctx context.Context,
+	p vrfPipelineResult,
+	fromAddress common.Address,
+) (etx txmgr.EvmTx, err error) {
+	if lsn.job.VRFSpec.VRFOwnerAddress == nil {
+		err = errors.New("vrf owner address not set in job spec, recreate job and provide it to force-fulfill")
+		return
+	}
+
+	if p.payload == "" {
+		// should probably never happen
+		// a critical log will be logged if this is the case in simulateFulfillment
+		err = errors.New("empty payload in vrfPipelineResult")
+		return
+	}
+
+	// fulfill the request through the VRF owner
+	err = lsn.q.Transaction(func(tx pg.Queryer) error {
+		if err = lsn.logBroadcaster.MarkConsumed(p.req.lb, pg.WithQueryer(tx)); err != nil {
+			return err
+		}
+
+		lsn.l.Infow("VRFOwner.fulfillRandomWords vs. VRFCoordinatorV2.fulfillRandomWords",
+			"vrf_owner.fulfillRandomWords", hexutil.Encode(vrfOwnerABI.Methods["fulfillRandomWords"].ID),
+			"vrf_coordinator_v2.fulfillRandomWords", hexutil.Encode(coordinatorV2ABI.Methods["fulfillRandomWords"].ID),
+		)
+
+		vrfOwnerAddress1 := lsn.vrfOwner.Address()
+		vrfOwnerAddressSpec := lsn.job.VRFSpec.VRFOwnerAddress.Address()
+		lsn.l.Infow("addresses diff", "wrapper_address", vrfOwnerAddress1, "spec_address", vrfOwnerAddressSpec)
+
+		txData, err := vrfOwnerABI.Pack("fulfillRandomWords", p.proof, p.reqCommitment)
+		if err != nil {
+			return errors.Wrap(err, "abi pack VRFOwner.fulfillRandomWords")
+		}
+		estimateGasLimit, err := lsn.ethClient.EstimateGas(ctx, ethereum.CallMsg{
```

Comment:
can be more bullet-proof if we apply a multiplier to `estimateGasLimit`. I believe it's 1.1 for most chains but can be more aggressive. That being said, I don't think it's a blocker since the occurrence of force fulfill should be very rare and we can always manually fulfill.

Reply:
Will add a multiplier

Reply:
The problem is that the multiplier is not part of the spec at the moment, so we'd have to either add one or hardcode it. Not a fan of either approach.